app/classifier/drawing-tools/column-rectangle.cjsx

```diff
@@ -0,0 +1,81 @@
+React = require 'react'
+DrawingToolRoot = require './root'
+DragHandle = require './drag-handle'
+Draggable = require '../../lib/draggable'
+DeleteButton = require './delete-button'
+
+DELETE_BUTTON_DISTANCE = 9 / 10
+DEFAULT_WIDTH = 25
+MINIMUM_WIDTH = 25
+
+module.exports = React.createClass
+  displayName: 'RectangleTool'
+
+  statics:
+    initCoords: null
+
+    defaultValues: ({x, y}) ->
+      x: x
+      y: 0
+      width: DEFAULT_WIDTH
+
+    initStart: ({x, y}, mark) ->
+      @initCoords = {x, 0}
+      {x, 0, _inProgress: true}
+
+    initMove: (cursor, mark) ->
+      if cursor.x > @initCoords.x
+        width = cursor.x - mark.x
+        x = mark.x
+      else
+        width = @initCoords.x - cursor.x
+        x = cursor.x
+
+      y = mark.y
+
+      {x, y, width}
+
+    initRelease: ->
+      _inProgress: false
+
+    initValid: (mark) ->
+      mark.width >= MINIMUM_WIDTH
+
+  initCoords: null
+
+  render: ->
+    {x, y, width} = @props.mark
+    allowedWidth = if @props.mark.width < MINIMUM_WIDTH then MINIMUM_WIDTH else width
```

Comment:
This decouples the display of the mark from the values in the annotation. Enforce the minimum in the collected data, not the rendered output.